### PR TITLE
feat(cb2-6731): desk based testing template 1

### PR DESCRIPTION
**Pull request by kpearmanbjss**
Implements the template for the desk based testing for a PSV Group 1

[link to ticket number](https://dvsa.atlassian.net/browse/CB2-6731)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by Ruairidh-BJSS — APPROVED** [review]

**Review by gjulien-bjss — DISMISSED** [review]

**Review by mcadenas-bjss — APPROVED** [review]

File: src/app/forms/validators/custom-validators.ts [hunk 1]
```diff
@@ -167,11 +167,10 @@ export class CustomValidators {
 
   static aheadOfDate = (sibling: string): ValidatorFn => {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control?.parent) {
-        const siblingControl = control.parent.get(sibling) as CustomFormControl;
-
+      const siblingControl = control?.parent?.get(sibling);
+      if (siblingControl?.value) {
```
Comment (tom-bjss) [inline src/app/forms/validators/custom-validators.ts:171]:
```suggestion
      if (siblingControl?.value && new Date(control.value) < new Date(siblingControl.value)) {
```

File: src/app/forms/validators/custom-validators.ts [hunk 2]
```diff
@@ -181,13 +180,13 @@ export class CustomValidators {
...
   static dateNotExceed = (sibling: string, months: number): ValidatorFn => {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control?.parent) {
-        const siblingControl = control.parent.get(sibling) as CustomFormControl;
+      const siblingControl = control?.parent?.get(sibling);
+      if (siblingControl?.value) {
         const maxDate = new Date(siblingControl.value);
         maxDate.setMonth(maxDate.getMonth() + months);
 
         if (new Date(control.value) > maxDate) {
```
Comment (tom-bjss) [inline src/app/forms/validators/custom-validators.ts:188]: same here

**Review by Ruairidh-BJSS — DISMISSED** [review]

**Review by tom-bjss — COMMENTED** [review]
feat(cb2-6731): code cleanup

File: src/app/forms/validators/custom-validators.ts [hunk 3]
```diff
@@ -181,13 +178,13 @@ export class CustomValidators {
 
   static dateNotExceed = (sibling: string, months: number): ValidatorFn => {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control?.parent) {
-        const siblingControl = control.parent.get(sibling) as CustomFormControl;
+      const siblingControl = control?.parent?.get(sibling);
+      if (siblingControl?.value) {
```
Comment (mcadenas-bjss) [inline src/app/forms/validators/custom-validators.ts:182]: Should we not also check `control.value` to be truthy?

**Review by Ruairidh-BJSS — APPROVED** [review]

**Pull request merged**